### ISSUE TITLE
ViewInspector inspects "destination" property of NavigationLink, can recurse infinitely

### ISSUE THREAD

**Issue by varun531** [nalexn/ViewInspector#165]
Using ViewInspector 0.9.1, XCode 13.2.1.

If I have the following views:

```swift

struct ViewA: View {
    var body: some View {
        Text("viewA")
        NavigationLink("Goto view B", destination: ViewB())
    }
}

struct ViewB: View {
    var body: some View {
        Text("viewB")
    }
}

```

and I try to search for the text "viewB" on `ViewA` via:

```swift
let sut = ViewA()
XCTAssertNotNil(try sut.inspect().find(text: "viewB"))
```

This test will actually pass for me, because ViewInspector treats the `NavigationLink` as a child of `ViewA`, and its `destination` as a child of the `NavigationLink`.

NOTE: If `ViewB` has a `NavigationLink` back to `ViewA`, ViewInspector will recurse infinitely while inspecting `ViewA`.

**Comment by varun531** [comment]
@Tyler-Keith-Thompson it should NOT be inspecting the `destination` property of the `NavigationLink` when inspecting `ViewA`. Inspection/verification of `ViewA` content is meaningless if it ends up inspecting ALL views that can be reached from `ViewA` via `NavigationLink`s.

**Comment by varun531** [comment]
@Tyler-Keith-Thompson I agree that having a configurable recursion depth is bad, because it limits arbitrarily nested views. I would be happy having `find` skip over destinations of inactive `NavigationLink`s. Regardless, I hope you agree that the scenario I posted above is unacceptable, because it seriously screws up the signal of the tests.

**Comment by nalexn** [comment]
Yeah, I see. I suspect the correct `find` behavior should be to continue the search in the `NavigationLinks`'s destination only if it is active. I'll look into a fix

**Comment by nalexn** [comment]
I've added a fix to both infinite recursion and the destination view being accessible for inactive links. FYI, for the latter, the check is only employed for NavigationLinks with `Binding` parameter for "selection". Without an explicit `selection` parameter, it is currently impossible to reliably identify active / inactive links, so it falls back to the old behavior, that is to access the destination regardless of the link selection state.

**Comment by nalexn** [comment]
Fixed released with v0.9.4
